client/packages/programs/src/JsonForms/common/components/Text.tsx

```diff
@@ -10,26 +10,33 @@ import { FORM_LABEL_WIDTH } from '../styleConstants';
 import { z } from 'zod';
 import { useZodOptionsValidation } from '../useZodOptionsValidation';
 
-type Options = {
-  /**
-   * Additional pattern to be matched that can be defined in ui schema
-   */
-  pattern?: string;
-  /**
-   * Examples for the correct pattern
-   */
-  examples?: string[];
-  width?: string;
-};
-const Options: z.ZodType<Options | undefined> = z
+const Options = z
   .object({
+    /**
+     * Additional pattern to be matched that can be defined in ui schema
+     */
     pattern: z.string().optional(),
+    /**
+     * Examples for the correct pattern
+     */
     examples: z.array(z.string()).optional(),
     width: z.string().optional(),
+    /**
+     * If true, text input will expand to multiple lines if required (default:
+     * true)
+     */
+    multiline: z.boolean().optional(),
+    /**
+     * How many rows should the textbox display initially (default: 1, ignored
+     * if `multiline === false`)
+     */
+    rows: z.number().optional(),
```

Comment:
cool! thanks for trying it out. I have to say I find the zod typedefinitions much harder to read though. But I guess its ok if you hover it in the IDE.